### PR TITLE
generate pojo beans from openapi document

### PR DESCRIPTION
**Pull request by techcobweb**
Signed-off-by: Mike Cobbett <77053+techcobweb@users.noreply.github.com>

This relates to this story: https://github.com/galasa-dev/projectmanagement/issues/529

- Generates a java client from the openapi yaml file.
- Includes generating a set of transport beans (in the 'model' package).
- Requires some changes to be done in the run-time, like wrapping all the dependencies and supplying them as OSGi bundles.
- Builds and compiles cleanly

DO NOT MERGE YET, ...until we have:
- Decided whether the benefit of doing this outweighs the effort of adopting these POJOs
- Added the dependencies in wrappings
- Trialed their use in one of the servlets

### CODE REVIEW

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

File: galasa-parent/dev.galasa.framework.api.client/build.gradle [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    id 'galasa.api.server'
+    id "org.openapi.generator" version "6.6.0"
+
+    // testCompile requires Java plugin.
+    id 'java'
+}
+
+description = 'Framework API - REST java client'
+
+version = '0.31.0'
```
Comment (eamansour) [inline galasa-parent/dev.galasa.framework.api.client/build.gradle:12]: Should this be 0.32.0 now?

File: galasa-parent/settings.gradle [hunk 2]
```diff
@@ -22,6 +22,7 @@ include 'dev.galasa.framework.metrics'
 include 'dev.galasa.framework.api'
 include 'dev.galasa.framework.api.authentication'
 include 'dev.galasa.framework.api.bootstrap'
+include 'dev.galasa.framework.api.client'
```
Comment (eamansour) [inline galasa-parent/settings.gradle:25]: We likely need `dev.galasa.framework.api.client` in the framework's release.yaml as well

**Review by eamansour — APPROVED** [review]
A couple of minor comments but approved.

File: galasa-parent/dev.galasa.framework.api.client/build.gradle (hunk 1 above)
Comment (techcobweb) [reply galasa-parent/dev.galasa.framework.api.client/build.gradle:12]: yep.

Comment (techcobweb) [reply galasa-parent/dev.galasa.framework.api.client/build.gradle]: Fixed

File: galasa-parent/settings.gradle (hunk 2 above)
Comment (techcobweb) [reply galasa-parent/settings.gradle:25]: Ok. Added.

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Pull request merged**